README


This archive contains the distribution Plack-App-BeanstalkConsole,
version 0.007:

  a web application that provides access to Beanstalk statistics and tools

This software is copyright (c) 2013 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.




Changes
Revision history for Plack-App-BeanstalkConsole

0.007     2013-12-10 20:08:34Z
          - now properly dying in build, rather than exiting with status 0, if
            required php-cgi executable is not available

0.006     2013-12-08 04:40:53Z
          - declare extra module used in Makefile.PL as configure_requires
          - now also shipping a Build.PL, with Makefile.PL serving as a
            fallback

0.005     2013-06-05 17:22:59 CDT-0500 (Karen Etheridge)
          - installer bails out if php-cgi is not installed

0.004     2013-05-27 12:07:05 PDT-0700 (Karen Etheridge)
          - fix '/' URI redirection - direct mapping confuses CSS

0.003     2013-05-27 11:23:03 PDT-0700 (Karen Etheridge)
          - the '/' URI is now mapped to '/public/', to avoid a bit of newbie
            confusion

0.002     2013-05-26 17:32:41 PDT-0700 (Karen Etheridge)
          - inc::DownloadShareDirContent has been rewritten to not patch
            [MakeMaker] directly, but rather make use of [MakeMaker::Awesome].

0.001     2013-05-26 10:00:07 PDT-0700 (Karen Etheridge)
          - Initial release.



Build.PL
exit 0 if $ENV{AUTOMATED_TESTING};
use File::Which;
die "php-cgi not found; aborting" if not which("php-cgi");
# begin inc::DownloadShareDirContent (1)
use File::Spec;
use File::Temp 'tempdir';
use HTTP::Tiny;
use Archive::Extract;

my $archive_file = File::Spec->catfile(tempdir(CLEANUP => 1), "beanstalk_console.zip");
print "downloading https://github.com/downloads/ptrofimov/beanstalk_console/beanstalk_console.zip to $archive_file...
";
my $response = HTTP::Tiny->new->mirror('https://github.com/downloads/ptrofimov/beanstalk_console/beanstalk_console.zip', $archive_file);
$response->{success} or die "failed to download https://github.com/downloads/ptrofimov/beanstalk_console/beanstalk_console.zip into $archive_file";

my $extract_dir = 'share';
my $ae = Archive::Extract->new(archive => $archive_file);
$ae->extract(to => $extract_dir) or die "failed to extract $archive_file to $extract_dir ";
# end inc::DownloadShareDirContent (1)
use 5.006;
use Module::Build::Tiny 0.030;
Build_PL();


lib/Plack/App/BeanstalkConsole.pm
use strict;
use warnings;
package Plack::App::BeanstalkConsole;
{
  $Plack::App::BeanstalkConsole::VERSION = '0.007';
}
# git description: v0.006-2-g45485a2

BEGIN {
  $Plack::App::BeanstalkConsole::AUTHORITY = 'cpan:ETHER';
}
# ABSTRACT: a web application that provides access to Beanstalk statistics and tools

use parent 'Plack::App::PHPCGIFile';

use File::ShareDir;
use Scalar::Util 'blessed';
use namespace::autoclean;

sub prepare_app
{
    my $self = shift;
    if (not $self->{root})
    {
        my $class = blessed $self;
        (my $dist = $class) =~ s/::/-/g;
        $self->{root} = File::ShareDir::dist_dir($dist);
    }
    $self->SUPER::prepare_app;
}

sub call
{
    my ($self, $env) = @_;

    # / -> /public/
    # CSS is screwed up if we rewrite PATH_INFO directly here.
    return [ '301', [ 'Location' => '/public/' ], [] ]
        if $env->{PATH_INFO} eq '/';

    # */ -> */index.php
    $env->{PATH_INFO} .= 'index.php'
        if substr($env->{PATH_INFO}, -1, 1) eq '/';

    $self->SUPER::call($env);
}

1;

__END__

=pod

=encoding UTF-8

=for :stopwords Karen Etheridge Petr Sergey Trofimov irc beanstalkd

=head1 NAME

Plack::App::BeanstalkConsole - a web application that provides access to Beanstalk statistics and tools

=head1 VERSION

version 0.007

=head1 SYNOPSIS

    use Plack::App::BeanstalkConsole;
    # accessible under /...
    my $app = Plack::App::BeanstalkConsole->new->to_app;

    # Or mount on a specific path
    use Plack::Builder;
    builder {
        # accessible under /beanstalk/...
        mount beanstalk => Plack::App::BeanstalkConsole->new;
    };

See L<plackup> for how to quickly and easily mount this application from the
command line.

=head1 DESCRIPTION

This is a simple L<Plack> wrapper for the excellent
L<Beanstalk Console|https://github.com/ptrofimov/beanstalk_console>
application written in PHP by Петр Трофимов (Petr Trofimov)
and Сергей Лысенко (Sergey Lysenko).

The latest version of the application is downloaded at install time and saved
as a L<File::ShareDir|share dir>, which is used by default if the C<root> is
not overridden (see below).

To use, mount the app on your server and go to the '/' URI,
where you will be prompted to enter the address of your beanstalk server(s).

=head1 METHODS

=head2 C<new>

    Plack::App::BeanstalkConsole->new(<options>)

Options (passed as a hash):

=over 4

=item * C<root> (optional)

If not provided, the PHP code that was downloaded at install time is used.
However, you can override this option to point to any directory you wish, that
contains the PHP code to be mounted. (In this way it functions just like
L<Plack::App::PHPCGIFile>.)

    Plack::App::BeanstalkConsole->new(root => 'path/to/beanstalk_console')

=back

=head1 EXTERNAL REQUIREMENTS

The C<php-cgi> binary must be available in C<$PATH>.  In newer versions of
PHP, this is is normally installed as part of the main PHP installation.

=head1 SUPPORT

Bugs may be submitted through L<the RT bug tracker|https://rt.cpan.org/Public/Dist/Display.html?Name=Plack-App-BeanstalkConsole>
(or L<bug-Plack-App-BeanstalkConsole@rt.cpan.org|mailto:bug-Plack-App-BeanstalkConsole@rt.cpan.org>).
I am also usually active on irc, as 'ether' at C<irc.perl.org>.

=head1 SEE ALSO

=over 4

=item *

L<Plack>

=item *

L<Plack::App::PHPCGIFile>

=item *

L<Beanstalk Console|https://github.com/ptrofimov/beanstalk_console>

=item *

L<beanstalkd|http://kr.github.com/beanstalkd>

=back

=head1 AUTHOR

Karen Etheridge <ether@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2013 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL
exit 0 if $ENV{AUTOMATED_TESTING};
use File::Which;
die "php-cgi not found; aborting" if not which("php-cgi");
# This Makefile.PL for Plack-App-BeanstalkConsole was generated by
# inc::DownloadShareDirContent <self>.
# Don't edit it but the dist.ini used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'File::Spec' => '0',
    'Module::Build::Tiny' => '0.030',
    'File::ShareDir::Install' => '0.03',
    'Archive::Extract' => '0',
    'HTTP::Tiny' => '0',
    'ExtUtils::MakeMaker' => '6.30',
    'File::Which' => '0',
    'File::Temp' => '0',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;

use ExtUtils::MakeMaker 6.30;

use File::ShareDir::Install;
# begin inc::DownloadShareDirContent (1)
use File::Spec;
use File::Temp 'tempdir';
use HTTP::Tiny;
use Archive::Extract;

my $archive_file = File::Spec->catfile(tempdir(CLEANUP => 1), "beanstalk_console.zip");
print "downloading https://github.com/downloads/ptrofimov/beanstalk_console/beanstalk_console.zip to $archive_file...
";
my $response = HTTP::Tiny->new->mirror('https://github.com/downloads/ptrofimov/beanstalk_console/beanstalk_console.zip', $archive_file);
$response->{success} or die "failed to download https://github.com/downloads/ptrofimov/beanstalk_console/beanstalk_console.zip into $archive_file";

my $extract_dir = 'share';
my $ae = Archive::Extract->new(archive => $archive_file);
$ae->extract(to => $extract_dir) or die "failed to extract $archive_file to $extract_dir ";
# end inc::DownloadShareDirContent (1)

# begin inc::DownloadShareDirContent (2)
install_share dist => $extract_dir;
# end inc::DownloadShareDirContent (2)


my %WriteMakefileArgs = (
  "ABSTRACT" => "a web application that provides access to Beanstalk statistics and tools",
  "AUTHOR" => "Karen Etheridge <ether\@cpan.org>",
  "BUILD_REQUIRES" => {},
  "CONFIGURE_REQUIRES" => {
    "Archive::Extract" => 0,
    "ExtUtils::MakeMaker" => "6.30",
    "File::ShareDir::Install" => "0.03",
    "File::Spec" => 0,
    "File::Temp" => 0,
    "File::Which" => 0,
    "HTTP::Tiny" => 0,
    "Module::Build::Tiny" => "0.030"
  },
  "DISTNAME" => "Plack-App-BeanstalkConsole",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Plack::App::BeanstalkConsole",
  "PREREQ_PM" => {
    "File::ShareDir" => 0,
    "Plack::App::PHPCGIFile" => 0,
    "Scalar::Util" => 0,
    "namespace::autoclean" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec::Functions" => 0,
    "HTTP::Request::Common" => 0,
    "List::Util" => 0,
    "Plack::Test" => 0,
    "Test::File::ShareDir" => 0,
    "Test::More" => 0,
    "Test::Warnings" => 0
  },
  "VERSION" => "0.007",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  my $tr = delete $WriteMakefileArgs{TEST_REQUIRES};
  my $br = $WriteMakefileArgs{BUILD_REQUIRES};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

unless ( eval { ExtUtils::MakeMaker->VERSION(6.56) } ) {
  my $br = delete $WriteMakefileArgs{BUILD_REQUIRES};
  my $pp = $WriteMakefileArgs{PREREQ_PM};
  for my $mod ( keys %$br ) {
    if ( exists $pp->{$mod} ) {
      $pp->{$mod} = $br->{$mod} if $br->{$mod} > $pp->{$mod};
    }
    else {
      $pp->{$mod} = $br->{$mod};
    }
  }
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

{
package
MY;
use File::ShareDir::Install qw(postamble);
}


